src/sentry/buffer/redis.py

```diff
@@ -97,24 +98,26 @@ def get_routing_client(self) -> RedisCluster | rb.RoutingClient:
         else:
             raise AssertionError("unreachable")
 
-    def validate(self):
+    def validate(self) -> None:
         validate_dynamic_cluster(self.is_redis_cluster, self.cluster)
 
-    def _coerce_val(self, value):
+    def _coerce_val(self, value: models.Model | str | int) -> bytes:
         if isinstance(value, models.Model):
             value = value.pk
         return force_bytes(value, errors="replace")
 
-    def _make_key(self, model, filters):
+    def _make_key(
+        self, model: type[models.Model], filters: dict[str, models.Model | str | int]
+    ) -> str:
         """
         Returns a Redis-compatible key for the model given filters.
         """
         md5 = md5_text(
-            "&".join(f"{k}={self._coerce_val(v)}" for k, v in sorted(filters.items()))
+            "&".join(f"{k}={self._coerce_val(v)}" for k, v in sorted(filters.items()))  # type: ignore[str-bytes-safe]
```

Comment:
instead of `type: ignore` this should be `{self._corece_val(v)!r}` to be explicit that we're using repr

Reply:
This feels "python magic". Thanks!

Reply:
ah yeah -- in string formatting `{o!r}` is shorthand for `{repr(o)}` -- if an object does not implement `__str__` it defaults to `__repr__` -- and formatting a bytes object is usually a mistake (it may have been here -- or a leftover from python 2 -- but the mistake doesn't really impact anything since it's just use to generate a hash)